src/blockdata/locktime/relative.rs

```diff
@@ -0,0 +1,155 @@
+///
+/// ### Relevant BIPs
+///
+/// * [BIP 68 Relative lock-time using consensus-enforced sequence numbers](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki)
+/// * [BIP 112 CHECKSEQUENCEVERIFY](https://github.com/bitcoin/bips/blob/master/bip-0112.mediawiki)
+#[allow(clippy::derive_ord_xor_partial_ord)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub enum LockTime {
+    /// A block height lock time value.
+    Blocks(Height),
+    /// A 512 second time interval value.
+    Time(Time),
+}
+
+impl LockTime {
+    /// Creates a [`relative::LockTime`], locked by height, from `value`.
+    pub(crate) fn from_height(value: u16) -> Self {
+        LockTime::Blocks(Height(value))
+    }
+
+    /// Creates a [`relative::LockTime`], locked by time intervals, from `value`.
+    pub(crate) fn from_time(value: u16) -> Self {
+        LockTime::Time(Time(value))
+    }
+
+    /// Returns true if this [relative::`LockTime`] is satisfied by sequence number `n`.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitcoin::{relative, Sequence};
+    ///
+    /// let height: u16 = 100;
+    /// let lock = Sequence::from_height(height).to_relative_lock_time().expect("valid height");
+    /// assert!(lock.is_satisfied_by(Sequence::from_height(height+1)));
+    /// ```
+    // FIXME: Have I got the terminology around the correct way? Does self satisfy n or n satisfy self?
+    pub fn is_satisfied_by(&self, n: Sequence) -> bool {
+        if !n.is_relative_lock_time() {
+            return false;
+        }
+
+        match *self {
+            LockTime::Blocks(height) => match n.height_value() {
+                Some(value) => height.value() <= value,
+                None => false,
+            }
+            LockTime::Time(time) => match n.time_value() {
+                Some(value) => time.value() <= value,
+                None => false,
+            }
+        }
+    }
+
+    /// Returns true if this [relative::`LockTime`] is satisfied by `n`.
+    ///
+    /// Only the bottom 16 bits are used as the value of a relative time lock (see [BIP 68]).
+    ///
+    /// # Returns
+    ///
+    /// Returns false if `self` is not a height based lock or if `n` does not satisfy the lock.
+    ///
+    /// [BIP 68]: <https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki>
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitcoin::{relative, Sequence};
+    ///
+    /// let height: u16 = 100;
+    /// let lock = Sequence::from_height(height).to_relative_lock_time().expect("valid height");
+    /// assert!(lock.is_satisfied_by_height(height+1));
+    /// ```
+    pub fn is_satisfied_by_height(&self, n: u16) -> bool {
+        use LockTime::*;
+
+        match *self {
+            Blocks(height) => height.0 <= n,
+            Time(_) => false,
```

Comment:
In 2e6a8aefc048c28f71d8a4efb5f1d2766354c9f2:

I think we shoud return `true` by default. Conceptually a height-based timelock has no constraint on time.

Reply:
Done.